Modules/_functoolsmodule.c

```diff
@@ -76,13 +76,12 @@ static PyMethodDef placeholder_methods[] = {
 };
 
 static void
-placeholder_dealloc(PyObject* placeholder)
+placeholder_dealloc(PyObject* self)
 {
-    /* This should never get called, but we also don't want to SEGV if
-     * we accidentally decref Placeholder out of existence. Instead,
-     * since Placeholder is an immortal object, re-set the reference count.
-     */
-    _Py_SetImmortal(placeholder);
+    PyObject_GC_UnTrack(self);
+    PyTypeObject *tp = Py_TYPE(self);
+    tp->tp_free((PyObject*)self);
```

Comment:
```suggestion
    tp->tp_free(self);
```
nit: there's no need to casting it to `PyObject *` since it's already a `PyObject *`